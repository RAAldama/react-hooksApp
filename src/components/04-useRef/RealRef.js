import { useState } from 'react';
import '../02-useEffect/SimpleForm.css';
import MultipleHooks from '../03-examples/MultipleHooks';

const RealRef = () => {
    const [show, setShow] = useState(false);

    const handleClick = () => {
        setShow(!show);
    }

    return (
        <div>
            <h1>Example UseRef</h1>
            <hr />

            {show && <MultipleHooks />}

            <button className="btn btn-outline-info mt-2" onClick={handleClick}>
                Mostrar/Ocultar
            </button>
        </div>
    )
}

export default RealRef
