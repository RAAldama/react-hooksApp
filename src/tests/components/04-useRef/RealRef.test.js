import { shallow } from "enzyme"
import RealRef from "../../../components/04-useRef/RealRef"

describe('tests on <RealRef />', () => {

    test('should show correctly', () => {
        const wrapper = shallow(<RealRef />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show the <MultipleHooks /> correctly', () => {
        const wrapper = shallow(<RealRef />);
        wrapper.find('button').simulate('click');

        expect( wrapper.find('MultipleHooks').exists() ).toBe(true);
    })
    
    
})
