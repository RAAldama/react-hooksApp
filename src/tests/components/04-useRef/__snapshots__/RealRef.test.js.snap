// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <RealRef /> should show correctly 1`] = `
<div>
  <h1>
    Example UseRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-info mt-2"
    onClick={[Function]}
  >
    Mostrar/Ocultar
  </button>
</div>
`;
